src/features/dashboard/components/PortfolioHistoryChart.tsx

```diff
@@ -1,115 +1,163 @@
-import { CartesianGrid, Line, LineChart, XAxis, YAxis } from 'recharts';
-import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
+import { Area, AreaChart, CartesianGrid, XAxis, YAxis } from 'recharts';
 import {
   ChartConfig,
   ChartContainer,
   ChartTooltip,
   ChartTooltipContent,
 } from '@/components/ui/chart';
-import { useState } from 'react';
-import { Button } from '@/components/ui/button';
-import { chartData } from '../mockData/portfolioHistoryData';
+import { usePortfolioPerformanceData } from '@/hooks/usePortflioPerformanceData/usePortfolioPerformanceData';
+import { HistoricalPriceData } from '@/types/historicalPrices';
+import { TimeRange } from './PortfolioHistoryChartCard';
+import { DailyPortfolio } from '@/utils/portfolioCalculations/getDailyPortfolio';
+
+type PortfolioHistoryChartProps = {
+  stockPrices: HistoricalPriceData[];
+  dailyPortfolio: DailyPortfolio;
+  selectedTimeRange: TimeRange;
+};
 
 const chartConfig = {
   portfolio_value: {
-    label: 'Portfolio value',
-    color: 'hsl(var(--chart-3))',
+    label: 'Hodnota portfolia',
+    color: 'hsl(var(--chart-2))',
   },
 } satisfies ChartConfig;
 
-type TimeRange = 'Za celou dobu' | '12M' | 'YTD' | '30D' | '7D';
-const timeRanges: TimeRange[] = ['Za celou dobu', '12M', 'YTD', '30D', '7D'];
+export const PortfolioHistoryChart = ({
+  stockPrices,
+  dailyPortfolio,
+  selectedTimeRange,
+}: PortfolioHistoryChartProps) => {
+  const { oneWeek, oneMonth, oneYear, ytd, allTime } =
+    usePortfolioPerformanceData(dailyPortfolio, stockPrices);
+
+  const chartData =
+    selectedTimeRange === 'Za celou dobu'
+      ? allTime
+      : selectedTimeRange === 'YTD'
+        ? ytd
+        : selectedTimeRange === '1M'
+          ? oneMonth
+          : selectedTimeRange === '1R'
+            ? oneYear
+            : oneWeek;
 
-export const PortfolioHistoryChart = () => {
-  const [selectedTimeRange, setSelectedTimeRange] = useState<TimeRange>('30D');
+  const formatXAxisTick = (dateStr: string, index: number) => {
+    const date = new Date(dateStr);
 
-  const handleTimeRangeChange = (range: TimeRange) => {
-    setSelectedTimeRange(range);
+    if (index === 0) {
+      return '';
+    }
+
+    switch (selectedTimeRange) {
+      case 'Za celou dobu':
+        return date.getFullYear().toString();
+      case '1R':
+      case 'YTD':
+        return (
+          date
+            .toLocaleDateString('cs-CZ', { month: 'short' })
+            .replace('.', '')
+            .charAt(0)
+            .toUpperCase() +
+          date
+            .toLocaleDateString('cs-CZ', { month: 'short' })
+            .replace('.', '')
+            .slice(1)
+        );
+      default:
+        return `${date.getDate()}.${date.getMonth() + 1}.`;
+    }
+  };
+
+  const getXAxisProps = () => {
+    switch (selectedTimeRange) {
```

Comment:
The same goes for this switch statement, an object map would be a better solution